consensus/ethash/algorithm.go

```diff
@@ -151,10 +151,10 @@ func generateCache(dest []uint32, epoch uint64, seed []byte) {
 		logFn("Generated ethash verification cache", "elapsed", common.PrettyDuration(elapsed))
 	}()
 	// Convert our destination slice to a byte buffer
-	header := *(*reflect.SliceHeader)(unsafe.Pointer(&dest))
+	header := (*reflect.SliceHeader)(unsafe.Pointer(&dest))
 	header.Len *= 4
```

Comment:
These statements are going to modify the len and cap of dest, which I don't think you want.

There's also a use of reflect.SliceHeader in ethash/ethash.go.

Reply:
What do you recommend is the appropriate fix? I'm not familiar with this bit of code or this usage of reflect.SliceHeader

Reply:
```
var cache []byte
cacheHdr := (*reflect.SliceHeader)(unsafe.Pointer(&cache))
dstHdr := (*reflect.SliceHeader)(unsafe.Pointer(&dst))
cacheHdr.Data = dstHdr.Data
cacheHdr.Len = dstHdr.Len * 4
cacheHdr.Cap = dstHdr.Cap * 4
```